platform/ui/src/components/InputDoubleRange/InputDoubleRange.tsx

```diff
@@ -0,0 +1,219 @@
+    updateRangeValues(values);
+  }, [values, minValue, maxValue]);
+
+  const LabelOrEditableNumber = (val, index) => {
+    return allowNumberEdit ? (
+      <InputNumber
+        minValue={minValue}
+        maxValue={maxValue}
+        value={val}
+        onChange={newValue => {
+          updateRangeValues(newValue, index);
+        }}
+        step={step}
+        labelClassName="text-white"
+        showAdjustmentArrows={showAdjustmentArrows}
+      />
+    ) : (
+      <span className={classNames(labelClassName ?? 'text-white')}>
+        {val}
+        {unit}
+      </span>
+    );
+  };
+
+  useEffect(() => {
+    return () => {
+      window.removeEventListener('mousemove', handleMouseMove);
+      window.removeEventListener('mouseup', handleGlobalMouseUp);
+    };
+  }, []);
+
+  const [selectedThumb, setSelectedThumb] = useState(null);
+
+  const handleMouseUp = () => {
+    setSelectedThumb(null);
+  };
+
+  const handleGlobalMouseUp = () => {
+    // Remove global mouse event listeners
+    window.removeEventListener('mousemove', handleMouseMove);
+    window.removeEventListener('mouseup', handleGlobalMouseUp);
+    setSelectedThumb(null); // Stop the dragging
+  };
+
+  const handleMouseDown = e => {
+    const rect = e.currentTarget.getBoundingClientRect(); // Use currentTarget instead of target
+    const x = e.clientX - rect.left;
+    const percentageClicked = (x / rect.width) * 100;
+
+    // Calculate the distances from the clicked point to both thumbs' positions
+    const distanceToStartThumb = Math.abs(percentageClicked - percentageStart);
+    const distanceToEndThumb = Math.abs(percentageClicked - percentageEnd);
+
+    // Check if the clicked point is within a threshold distance to either thumb
+    if (distanceToStartThumb < 10) {
+      setSelectedThumb(0);
+    } else if (distanceToEndThumb < 10) {
+      setSelectedThumb(1);
+    }
+
+    window.addEventListener('mousemove', handleMouseMove);
+    window.addEventListener('mouseup', handleGlobalMouseUp);
+  };
+
+  const handleMouseMove = e => {
+    if (selectedThumb !== null) {
+      const rect = e.currentTarget.getBoundingClientRect();
+      const x = e.clientX - rect.left;
+      const newValue =
+        Math.round(((x / rect.width) * (maxValue - minValue) + minValue) / step) * step;
+
+      // Make sure newValue is within [minValue, maxValue]
+      const clampedValue = Math.min(Math.max(newValue, minValue), maxValue);
+
+      // Ensure that left and right thumbs don't switch positions
+      if (selectedThumb === 0 && clampedValue >= rangeValue[1]) {
+        return;
+      }
+      if (selectedThumb === 1 && clampedValue <= rangeValue[0]) {
+        return;
+      }
+
+      // Update the correct values in the rangeValue array
+      const updatedRangeValue = [...rangeValue];
+      updatedRangeValue[selectedThumb] = clampedValue;
+      setRangeValue(updatedRangeValue);
+
+      // No need to call onChange here, since we're updating local state+
```

Comment:
I am confused by this comment...if there is no need to call onChange, then why are we doing it on the next line?